aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelFieldTypeConverter.java

```diff
@@ -0,0 +1,242 @@
+
+        SQLiteModelFieldTypeConverter nestedModelConverter =
+                new SQLiteModelFieldTypeConverter(nestedModelType, modelSchemaRegistry, gson);
+
+        Map<String, Object> mapForInnerModel = new HashMap<>();
+        for (Map.Entry<String, ModelField> entry : innerModelSchema.getFields().entrySet()) {
+            mapForInnerModel.put(entry.getKey(), nestedModelConverter.convertValueFromSource(cursor, entry.getValue()));
+        }
+        final String modelInJsonFormat = gson.toJson(mapForInnerModel);
+        try {
+            return gson.getAdapter(nestedModelType).fromJson(modelInJsonFormat);
+        } catch (IOException exception) {
+            LOGGER.warn("", exception);
+            return null;
+        }
+    }
+
+    private Object convertCustomTypeToTarget(Cursor cursor, ModelField field, int columnIndex) throws IOException {
+        final String stringValue = cursor.getString(columnIndex);
+        return gson.getAdapter(field.getType()).fromJson(stringValue);
+    }
+
+    private Object convertDateToTarget(Cursor cursor, ModelField field, int columnIndex) throws ParseException {
+        // TODO wire up the new Date/Time handling here
+        final String dateInStringFormat = cursor.getString(columnIndex);
+        if (dateInStringFormat != null) {
+            return SimpleDateFormat
+                    .getDateInstance()
+                    .parse(dateInStringFormat);
+        }
+        return null;
+    }
+
+    @Override
+    public Object convertValueFromTarget(Model model, ModelField field) throws DataStoreException {
+        final String fieldName = field.getName();
+
+        final Object fieldValue = getModelValue(model, fieldName);
+        if (fieldValue == null) {
+            return null;
+        }
+        final JavaFieldType javaFieldType = TypeConverter.getJavaFieldType(field);
+
+        switch (javaFieldType) {
+            case INTEGER:
+            case LONG:
+            case FLOAT:
+            case STRING:
+                // these types require no special treatment
+                return fieldValue;
+            case BOOLEAN:
+                boolean booleanValue = (boolean) fieldValue;
+                return booleanValue ? 1L : 0L;
+            case MODEL:
+                return ((Model) fieldValue).getId();
+            case ENUM:
+            case CUSTOM_TYPE:
+                return gson.toJson(fieldValue);
+            case DATE:
+                // TODO integrate with new Date/Time handling
+                final Date dateValue = (Date) fieldValue;
+                return SimpleDateFormat
+                        .getDateInstance()
+                        .format(dateValue);
+            case TIME:
+                return ((Time) fieldValue).getTime();
+            default:
+                LOGGER.warn("Field of type " + javaFieldType + " is not supported. Fallback to null.");
+                return null;
```

Comment:
What do you think about `gson.toJson(fieldValue)` as a default?

Reply:
I think that's a good idea, as long as we keep the `warn`

Reply:
I gave this some thought after a few tests and would like your opinion.

I believe we should fallback to `null` due to backward compatibility.

When the code reaches that `default` it means that a field type we should support is not correctly supported (i.e. we did not exhaust the enum). If we assume JSON, when that's fixed we are at risk of breaking backward compatibility with existing data in case the correct format isn't the same as the one produced by the `toJson()/fromJson()` call.

Does that make sense? Thoughts?

Reply:
Ah, yea I get what you're saying. Makes sense. `null` sounds safer.